### PR TITLE
feat: render custom emoji

### PR DESCRIPTION
**Pull request by account0123**
FastImage does not render svg file, then react-native-svg is imported.
`renderEmojis` function is called inside `spoilerRule` because if I create a new rule object there is a conflict.
### How renderEmojis work?
1. `renderEmojis` splits content by custom emoji regex
  i. Odd elements... are :ULID: then return as `CustomEmoji`
  ii. Even elements is splitted by default emoji regex (eg: `:smile:`) and mapped:
   - Odd elements are :name: then return as `SvgEmoji`
   - Even elements are strings

    iii. `subparts` flatmaps, each component is returned, each string is inspected for unicode characters search.

      1. If there are, characters are mapped as `SvgEmoji`, the array is interleaved with Text components and returns

    iv. Finally subparts (string|SvgEmoji array) is returned in `tokens` flat map.

2. At the end, elements can be a string array (no emojis found) or a string|Text|SvgEmoji|CustomEmoji array (at least one emoji found). 

## Please make sure to check the following tasks before opening and submitting a PR

* [X] I understand and have followed the [contribution guide](https://github.com/revoltchat/revolt/discussions/282)
* [X] I have tested my changes locally and they are working as intended
* [X] These changes do not have any notable side effects on other Revolt projects

### CODE REVIEW

File: src/lib/utils.ts [hunk 1]
```diff
@@ -99,6 +99,7 @@ export function parseRevoltNodes(text: string) {
     }
     return ping;
   });
+  /*
```
Comment (Rexogamer) [inline src/lib/utils.ts:102]: We can probably just remove this instead of commenting it out

File: src/lib/emojis.ts [hunk 2]
```diff
@@ -0,0 +1,2003 @@
+import {client} from '../Generic';
+export const emojiDictionary = {
```
Comment (Rexogamer) [inline src/lib/emojis.ts:2]: For this I was thinking of using [RevKit](https://github.com/Revolt-Unofficial-Clients/revkit), which comes with an emoji dictionary/various helper functions which can be found [here](https://github.com/Revolt-Unofficial-Clients/revkit/blob/master/core/src/utils/Emojis.ts) - this removes quite a bit of this burden from RVMob itself and places it on a community maintained and easily updatable package

**Review by Rexogamer — CHANGES_REQUESTED** [review]
Initial thoughts - I'll test this and clean up some other bits later, but let's start with these

**Comment by Rexogamer** [comment]
One other thing that came to mind: while I'm happy to add support for the various packs (something Revkit also helps with), I also want to give the user the ability to use their system's emoji font so we might want to add a new setting for this and not render them if said setting = "system"

File: src/lib/emojis.ts (hunk 2 above)
Comment (account0123) [reply src/lib/emojis.ts:2]: it is a good dependency indeed, i will give it a try

**Comment by Rexogamer** [comment]
Something I've also noticed is that messages take a lot longer to render and the client seems to disconnect/break more (presumably due to the amount of images it has to fetch); however, I'm not 100% sure how we could realistically solve this unless we bundle every emoji SVG with the app :thinking:

**Comment by Rexogamer** [comment]
(for external readers: we decided to comment out the built-in emoji renderer for now and just render custom emoji)

**Review by Rexogamer — APPROVED** [review]

**Pull request merged**